### PR TITLE
recycle reviews with rejected responses

### PR DESCRIPTION
**Pull request by LeoMcA**

### CODE REVIEW

File: mresponse/reviews/query.py [hunk 1]
```diff
@@ -7,15 +7,19 @@
 class ReviewQuerySet(models.QuerySet):
     def unresponded(self):
         """
-        Get reviews that have no response.
+        Get reviews that have no response, or only rejected responses.
         """
-        return self.filter(response__isnull=True)
+        return self.annotate(
+            responses_not_rejected_count=models.Count(
+                "responses", filter=models.Q(responses__rejected=False)
```
Comment (akatsoulas) [inline mresponse/reviews/query.py:14]: Why the `Q` object here?

File: mresponse/responses/models.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def assignment_expires_at(self):
 
 
 class Response(models.Model):
-    review = models.OneToOneField(
-        "reviews.Review", models.PROTECT, related_name="response"
+    review = models.ForeignKey(
+        "reviews.Review", models.PROTECT, related_name="responses"
```
Comment (akatsoulas) [inline mresponse/responses/models.py:32]: Does it make sense to have this field nullable? Also should this be changed to cascade?

File: mresponse/reviews/query.py (hunk 1 above)
Comment (LeoMcA) [reply mresponse/reviews/query.py:14]: I believe it's required to filter within the aggregation

File: mresponse/responses/models.py (hunk 2 above)
Comment (LeoMcA) [reply mresponse/responses/models.py:32]: I don't think so, for it to be a response, it must be responding to something (and hence have a non-NULL review).

I'm split on the cascade question, everything else seems to be PROTECTed in respond. Whatever you think best.

Comment (akatsoulas) [reply mresponse/responses/models.py:32]: I mean if you delete a review similar to the spirit of replying to none, it does make sense in mind to delete the responses of the review too. The opposite though should not happen. Responses can be deleted without affecting the review

Comment (LeoMcA) [reply mresponse/responses/models.py:32]: Absolutely, but PROTECT makes it impossible to delete the review if it has any responses, right? I'm not sure if we should be deleting reviews at all?

Comment (akatsoulas) [reply mresponse/responses/models.py:32]: We shouldn't but sometimes is needed - Eg staging comes to me. Anyways, I have not any strong feelings about this.

File: mresponse/responses/models.py [hunk 3]
```diff
@@ -28,8 +28,8 @@ def assignment_expires_at(self):
 
 
 class Response(models.Model):
-    review = models.OneToOneField(
-        "reviews.Review", models.PROTECT, related_name="response"
+    review = models.ForeignKey(
+        "reviews.Review", models.CASCADE, related_name="responses"
```
Comment (akatsoulas) [inline mresponse/responses/models.py:32]: Going a full circle on this, I believe your earlier comments made a good point and maybe we shall stick with `PROTECT` :D

File: mresponse/reviews/query.py [hunk 4]
```diff
@@ -7,15 +7,15 @@
 class ReviewQuerySet(models.QuerySet):
     def unresponded(self):
         """
-        Get reviews that have no response.
+        Get reviews that have no response, or only rejected responses.
         """
-        return self.filter(response__isnull=True)
+        return self.exclude(responses__rejected=False)
```
Comment (akatsoulas) [inline mresponse/reviews/query.py:12]: that looks a tiny bit better :P

**Pull request merged**